src/raygun.rum/index.js

```diff
@@ -117,33 +117,39 @@ var raygunRumFactory = function (window, $, Raygun) {
         Raygun.CoreWebVitals.attach(sendCoreWebVitalTimings, self.parentResource);
       }
 
-      var clickHandler = function () {
-        this.updateStorageTimestamp();
-      }.bind(_private);
-
-      var unloadHandler = function () {
+      var sendAllQueuedPayloadsWithBeacon = function () {
         self.sendUsingNavigatorBeacon = true;
         sendChildAssets(true);
         sendQueuedItems();
       }.bind(_private);
 
+      var clickHandler = function () {
+        this.updateStorageTimestamp();
+      }.bind(_private);
+
       var visibilityChangeHandler = function () {
         if (document.visibilityState === 'visible') {
           this.updateStorageTimestamp();
         }
+
+        if (document.visibilityState === 'hidden') {
+          sendAllQueuedPayloadsWithBeacon();
+        }
+      }.bind(_private);
+
+      var urlChangeHandler = function () {
+        sendAllQueuedPayloadsWithBeacon();
       }.bind(_private);
 
       var pageHideHandler = function () {
-        self.sendUsingNavigatorBeacon = true;
-        sendChildAssets(true);
-        sendQueuedItems();
+        sendAllQueuedPayloadsWithBeacon();
         self.stopCollectingMetrics = true;
       }.bind(_private);
 
       if (window.addEventListener) {
         window.addEventListener('click', clickHandler);
         document.addEventListener('visibilitychange', visibilityChangeHandler);
-        window.addEventListener('beforeunload', unloadHandler);
+        window.addEventListener('popstate', urlChangeHandler);
```

Comment:
https://caniuse.com/?search=window%20popstate

Looks like this is widely supported so don't see any issues.